mapboxgl/layers.py

```diff
@@ -0,0 +1,745 @@
+            strokeColor=self.stroke_color,
+            labelColor=self.label_color,
+            labelSize=self.label_size,
+            labelHaloColor=self.label_halo_color,
+            labelHaloWidth=self.label_halo_width
+        ))
+
+
+class HeatmapLayer(AbstractLayer):
+    """Create a heatmap layer"""
+
+    def __init__(self,
+                 data,
+                 label_property=None,
+                 weight_property=None,
+                 weight_stops=None,
+                 color_stops=None,
+                 radius_stops=None,
+                 intensity_stops=None,
+                 *args,
+                 **kwargs):
+        """Create a heatmap layer
+
+        :param weight_property: property to determine heatmap weight. EX. "population"
+        :param weight_stops: stops to determine heatmap weight.  EX. [[10, 0], [100, 1]]
+        :param color_stops: stops to determine heatmap color.  EX. [[0, "red"], [0.5, "blue"], [1, "green"]]
+        :param radius_stops: stops to determine heatmap radius based on zoom.  EX: [[0, 1], [12, 30]]
+        :param intensity_stops: stops to determine the heatmap intensity based on zoom. EX: [[0, 0.1], [20, 5]]
+
+        """
+        super(HeatmapLayer, self).__init__(data, *args, **kwargs)
+
+        self.template = 'heatmap'
+        self.label_property = label_property
+        self.weight_property = weight_property
+        self.weight_stops = weight_stops
+        if color_stops:
+            # Make the first color stop in a heatmap have opacity 0 for good visual effect
+            self.color_stops = [[0.00001, 'rgba(0,0,0,0)']] + color_stops
+        self.radius_stops = radius_stops
+        self.intensity_stops = intensity_stops
+
+    def get_layer_specific_variables(self, options):
+        """Update map template variables specific to heatmap visual"""
+        options.update(dict(
+            geojson_data=json.dumps(self.data, ensure_ascii=False),
+            colorStops=self.color_stops,
+            radiusStops=self.radius_stops,
+            weightProperty=self.weight_property,
+            weightStops=self.weight_stops,
+            intensityStops=self.intensity_stops,
+        ))
+
+
+class ClusteredCircleLayer(AbstractLayer):
+    """Create a clustered circle layer"""
+
+    def __init__(self,
+                 data,
+                 label_property=None,
+                 label_size=8,
+                 label_color='#131516',
+                 label_halo_color='white',
+                 label_halo_width=1,
+                 color_stops=None,
+                 radius_stops=None,
+                 cluster_radius=30,
+                 cluster_maxzoom=14,
+                 radius_default=2,
+                 color_default='black',
+                 stroke_color='grey',
+                 stroke_width=0.1,
+                 legend_key_shape='circle',
+                 *args,
+                 **kwargs):
+        """Construct a Mapviz object
```

Comment:
Docstring seems to be misleading. Why not `Construct ClusteredCircleLayer object` ?

Reply:
Good catch, I still need to go to all the docstrings and update them.